### ISSUE TITLE
Does not work on Fedora 30 (Wayland)

### ISSUE THREAD

**Issue by rgeary1** [pavelfatin/typometer#8]
Firstly, thanks for making this *excellent and unique* tool!

I've been unable to run this succesfully on Fedora 30 with the Gnome 3 window manager.  The generated key presses do not appear in gnome-terminal or other apps, but they do appear when I benchmark my IDEA ide.  Furthermore, even when the generated characters do appear, they aren't detected by the app.  Lastly, the mouse isn't able to click on anything after running the first benchmark.

The mouse click problem is fixed by using a newer JDK, I believe it is this bug :
https://bugs.openjdk.java.net/browse/JDK-8225118

By saving the screen to a file, I see the captured image is all black.  I think this is a XWayland issue, as I'm able to see a the screen capture when I log in with "Gnome on Xorg".  I wasn't able to create a Wayland solution, and I think the terminal in Gnome on Xorg is slower than Wayland (it reports ~35ms)

**Comment by bjonnh** [comment]
Yes Wayland applications are not able to spy on each other. Only the compositor can get access to all the buffers (and this is not standardized, there are at least two competing standards for that). In that case, until a support for your compositor is added, there is not much possible.
